Upgrade to JupyterLab 4 (#296)
jupyterlab/jupyter-ai: docs/source/users/index.md

## Prerequisites

You can run Jupyter AI on any system that can run a supported Python version
from 3.8 to 3.11, including recent Windows, macOS, and Linux versions.

If you use `conda`, you can install Python 3.11 in your environment by running:

```
conda install python=3.11
```

The `jupyter_ai` package, which provides the lab extension and user interface in
JupyterLab, depends on JupyterLab 4. If upgrading to JupyterLab 4 is not
possible in your environment, you can instead install `jupyter_ai` v1.x instead.
See "Installation" for more details.

JupyterLab may be installed two ways.

1. via `pip`:

```

> # change 4.0 to 3.0 if you need JupyterLab 3

Author: The version specifier `~=3.0` installs the latest version of JupyterLab 3. So running `pip install jupyterlab~=3.0` installs JupyterLab 3.6.x.

pip install jupyterlab~=4.0
```

2. via `conda`:

```

> # change 4.0 to 3.0 if you need JupyterLab 3

Reviewer: Ditto 3.6?

conda config --add channels conda-forge
conda config --set channel_priority strict
conda install jupyterlab~=4.0
```

The `jupyter_ai_magics` package, which provides exclusively the IPython magics,

> does not depend on JupyterLab or `jupyter_ai`. `jupyter_ai_magics` may be
> installed without `jupyterlab` or `jupyter_ai`.

Reviewer: ```suggestion
does not depend on JupyterLab or `jupyter_ai`. You can install `jupyter_ai_magics`
without installing `jupyterlab` or `jupyter_ai`.
```

## Installation

### Installation via `pip`

> If you want to install the JupyterLab extension, you can run:

Reviewer: ```suggestion
To install the JupyterLab extension, run:
```

```
pip install jupyter_ai
```

> The latest major version of `jupyter_ai` is currently v2, and supports
> exclusively JupyterLab 4.  If you need support for JupyterLab 3, you should use
> `jupyter_ai` v1 instead, which can be installed via:

Reviewer: ```suggestion
The latest major version of `jupyter_ai`, v2, only supports JupyterLab 4. 
If you need support for JupyterLab 3, you should install `jupyter_ai` v1 instead:
```

```
pip install jupyter_ai~=1.0
```

If you are not using JupyterLab and you only want to install the Jupyter AI `%%ai` magic, you can run:

```
$ pip install jupyter_ai_magics
```

`jupyter_ai` depends on `jupyter_ai_magics`, so installing `jupyter_ai`
automatically installs `jupyter_ai_magics`.

### Installation via `pip` within Conda environment (recommended)

We highly recommend installing both JupyterLab and Jupyter AI within an isolated
Conda environment to avoid clobbering Python packages in your existing Python
environment.

First, install
[conda](https://conda.io/projects/conda/en/latest/user-guide/install/index.html)
and create an environment that uses Python 3.11:

    $ conda create -n jupyter-ai python=3.11
    $ conda activate jupyter-ai
    $ pip install jupyter_ai

> Then, the steps from "Requirements" and "Installation via `pip`" may be followed
> to install JupyterLab and Jupyter AI within this Conda environment.

Reviewer: ```suggestion
Then, follow the steps from "Requirements" and "Installation via `pip`"
to install JupyterLab and Jupyter AI in this Conda environment.
```

Reviewer 2: Did clean install following steps of the potential user and found this part confusing. Following the page from top to bottom, I/user have already done Prerequisites step via pip or conda.

Following structure would be more clear:
1. pip only installation
    - Prerequisites installation
    - Jupyter AI installation
2. installation using conda environments 
    - Prerequisites installation
    - Jupyter AI installation

When starting JupyterLab with Jupyter AI, make sure to activate the Conda

> environment first:

Reviewer 2: I'd mention the need to make sure appropriate conda environment is active in the same spot conda is first mentioned on the page, in Prerequisites (or conda installation - Prerequisites if structure suggested in my comment above would be used)

```

> conda activate jupyter-ai

Reviewer: ```suggestion
conda activate jupyter-ai
jupyter lab
```

Should we include `jupyter lab` here?

Reviewer 2: +1 to @JasonWeill comment's spitit. We provide very detailed installation instructions, let's mention how to launch Jupyter AI as well

```

## Uninstallation

To remove the extension, run:

>     $ pip uninstall jupyter_ai

Reviewer: Should we use the triple-backtick syntax for code blocks in this section, for consistency with the sections above?

Author: This is outside the scope of the current PR.
